### PR TITLE
updated potential patient match ranking query connected to triage submit, without result display

### PR DESCRIPTION
**Pull request by elenarfowler**

### CODE REVIEW

File: app/femr/business/services/core/ISearchService.java [hunk 1]
```diff
@@ -108,8 +108,7 @@
      */
     ServiceResponse<List<PatientItem>> retrievePatientsFromQueryString(String patientSearchQuery);
 
-    //new search
-    ServiceResponse<List<PatientItem>> retrievePatientsFromTriageSearch(String first, String last);
+    ServiceResponse<List<PatientItem>> retrievePatientsFromTriageSearch(String first, String last, String phone, String addr, String gender, int age, String city);
```
Comment (rbhalodi) [inline app/femr/business/services/core/ISearchService.java]: thanks for expanding the search function!

File: app/femr/data/models/mysql/RankedPatientMatch.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package femr.data.models.mysql;
+import femr.data.models.core.IRankedPatientMatch;
+import io.ebean.annotation.Sql;
+import javax.persistence.*;
+
+@Entity
+@Sql
+public class RankedPatientMatch implements IRankedPatientMatch {
```
Comment (rbhalodi) [inline app/femr/data/models/mysql/RankedPatientMatch.java:8]: might want to add a test class for this later

File: public/js/triage/triage.js [hunk 3]
```diff
@@ -673,35 +675,23 @@ $(document).ready(function () {
...
-    // };
-
     function checkIfDuplicatePatientMatch() {
         var patientInfo = triageFields.patientInformation;
         var first = patientInfo.firstName.val();
         var last = patientInfo.lastName.val();
-        var url = "/search/findMatch/" + first + "/" + last;
+        var phone = patientInfo.phoneNumber.val();
+        var addr = patientInfo.address.val();
+        var age = patientInfo.years.val();
```
Comment (rbhalodi) [inline public/js/triage/triage.js]: we might need to fix the age later on due to the different age inputs and the way its calculated

File: app/femr/data/daos/system/PatientRepository.java [hunk 4]
```diff
@@ -233,6 +231,62 @@ public IPatient retrievePatientById(Integer id) {
...
+     */
+    @Override
+    public List<? extends IRankedPatientMatch> retrievePatientMatchesFromTriageFields(String firstName, String lastName, String phone, String addr, String gender, int age, String city) {
+
+        List<? extends IRankedPatientMatch> response = null;
+        try {
+
+            Query<? extends IRankedPatientMatch> query = QueryProvider.getRankedPatientMatchQuery();
+
+            String sql
```
Comment (rbhalodi) [inline app/femr/data/daos/system/PatientRepository.java:251]: the patient matching point system looks good

**Comment by rbhalodi** [comment]
Overall it looks great. I like how you expanded the new search function to take in additional fields and the ranking query looks good too, it follows the point system that we came up with earlier. The only thing is that the scala build fails, so maybe look into why that might be happening. Also, the new ranked matching class is a great addition, so you could add some test cases for it.

File: app/femr/data/models/mysql/RankedPatientMatch.java (hunk 2 above)
Comment (elenarfowler) [reply app/femr/data/models/mysql/RankedPatientMatch.java:8]: I added a Mock class that implements IRankedPatientMatch to be used in place of this class in tests.

File: public/js/triage/triage.js (hunk 3 above)
Comment (elenarfowler) [reply public/js/triage/triage.js]: I changed the search function to not compared the age number value but instead compare the date string (which is what is being stored in the db). I used the same converter function that already existed to get the date from age, then converted the date to type long to be passed over the router. Once I get the date as a long in the java code I convert it back to a date object then a string to be used in the ranking query.

**Pull request merged**